apps/dashboard/src/contract-ui/tabs/tokens/components/claim-button.tsx

```diff
@@ -1,37 +1,188 @@
+            <FormControl isRequired isInvalid={!!form.formState.errors.to}>
+              <FormLabel>To Address</FormLabel>
+              <Input
+                placeholder={ZERO_ADDRESS}
+                {...form.register("to", { required: true })}
+              />
+              <FormHelperText>Enter the address to claim to.</FormHelperText>
+              <FormErrorMessage>
+                {form.formState.errors.to?.message}
+              </FormErrorMessage>
+            </FormControl>
+            <FormControl isRequired isInvalid={!!form.formState.errors.amount}>
+              <FormLabel>Amount</FormLabel>
+              <Input
+                type="text"
+                pattern={`^\\d+(\\.\\d{1,${_decimals || 18}})?$`}
+                {...form.register("amount", { required: true })}
+              />
+              <FormHelperText>How many would you like to claim?</FormHelperText>
+              <FormErrorMessage>
+                {form.formState.errors.amount?.message}
+              </FormErrorMessage>
+            </FormControl>
+          </div>
+        </form>
+        <SheetFooter className="mt-10">
+          <TransactionButton
+            transactionCount={1}
+            form={CLAIM_FORM_ID}
+            isLoading={form.formState.isSubmitting}
+            type="submit"
+            colorScheme="primary"
+            isDisabled={!form.formState.isDirty || isLoading}
+            onClick={form.handleSubmit(async (d) => {
+              if (!d.to) {
+                return toast.error(
+                  "Need to speficy an address to receive tokens",
+                );
+              }
+              trackEvent({
+                category: "token",
+                action: "claim",
+                label: "attempt",
+              });
+              if (!account) {
+                return toast.error("No account detected");
+              }
+              const transaction = ERC20Ext.claimTo({
+                contract,
+                to: d.to,
+                quantity: d.amount,
+                from: account.address,
+              });
+
+              let approvedSuccessful = true;
+
+              const approveTx = await ERC20Ext.getApprovalForTransaction({
+                transaction,
+                account,
+              });
+
+              if (approveTx) {
+                const promise = sendAndConfirmTransaction.mutateAsync(
+                  approveTx,
+                  {
+                    onError: (error) => {
+                      console.error(error);
+                      approvedSuccessful = false;
+                    },
+                  },
+                );
+                toast.promise(promise, {
+                  loading: "Approving ERC20 tokens for this claim",
+                  success: "Tokens approved successfully",
+                  error: "Failed to approve token",
+                });
+
+                await promise;
```

Comment:
this should throw if it fails meaning you shouldn't need the separate `approvedSuccessful` - can easily test that by executing this and rejecting the approval TX in ur wallet, which should trigger the throw